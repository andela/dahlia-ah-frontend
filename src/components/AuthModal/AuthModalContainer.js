import React, { useState } from 'react';
import PropTypes from 'prop-types';
import AuthModal from './AuthModal';
import './AuthModal.scss';

const AuthModalContainer = ({
  children, title, desc, isOpen,
}) => {
  const [display, setDisplay] = useState(isOpen);
  const closeModal = () => {
    setDisplay(false);
  };

  return (
    <AuthModal title={title} desc={desc} isOpen={isOpen} display={display} closeModal={closeModal}>
      {children}
    </AuthModal>
  );
};

AuthModalContainer.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  title: PropTypes.string.isRequired,
  desc: PropTypes.string.isRequired,
  isOpen: PropTypes.bool.isRequired,
};

export default AuthModalContainer;
