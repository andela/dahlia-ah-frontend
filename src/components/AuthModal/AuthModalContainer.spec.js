import React from 'react';
import { configure, shallow, mount } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import AuthModalContainer from './AuthModalContainer';

configure({ adapter: new Adapter() });

describe('AuthModalContainer component <ResetPassword />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <AuthModalContainer />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders the AuthModal component', () => {
    const wrapper = shallow(
      <AuthModalContainer />,
    );

    expect(wrapper.find('AuthModal')).toHaveLength(1);
  });

  it('sets the display to none when a button is clicked', () => {
    const wrapper = mount(
      <AuthModalContainer isOpen />,
    );

    expect(wrapper.find('.auth-modal-div').get(0).props.style.display).toEqual('flex');
    wrapper.find('.fas').simulate('click');
    expect(wrapper.find('.auth-modal-div').get(0).props.style.display).toEqual('none');
    wrapper.unmount();
  });
});
