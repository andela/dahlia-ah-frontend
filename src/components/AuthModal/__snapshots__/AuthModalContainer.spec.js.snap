// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthModalContainer component <ResetPassword /> renders correctly 1`] = `ShallowWrapper {}`;
